python/tvm/relay/backend/contrib/tidl.py

```diff
@@ -0,0 +1,1403 @@
+        return None.
+
+        Parameters
+        ----------
+        mod : tvm.relay.Module 
+            Relay IR graph
+        params : dict of str to tvm.NDArray
+            The parameter dict to be used by relay
+        input: dictionary
+            A dictionary where the key in input name and the value is input tensor
+
+        Returns
+        -------
+        mod : tvm.relay.Module 
+            Paritioned graph with subgraphs to run with TIDL
+        """
+
+        if len(input) > 1:
+            print("TIDL currently only supports 1 input tensor");
+            return None
+        for key, value in input.items():
+            input_node = key
+            input_data = value
+
+        mod = relay.transform.RemoveUnusedFunctions()(mod)
+        # Bind params so that weights will appear as constants instead of variables 
+        mod['main'] = bind_params_by_name(mod['main'], params)
+        mod = relay.transform.FoldConstant()(mod)
+        mod['main'] = RemoveMultiplyByOne().visit(mod['main'])
+        print("---------- Original graph ----------")
+        print(mod.astext(show_meta_data=False))
+
+        #============= Annotate the graph ==============
+        # Looks at annotated ops and marks them in the graph with compiler.begin 
+        # and compiler.end.
+        # Merges annotated regions together that use the same external target, 
+        # and combines marked regions for each target
+        #Merge sequence of ops into composite functions/ops
+        print("---------- Merge Composite Functions ----------")
+        mod = tidl._merge_sequential_ops(mod) 
+        print("---------- Annotated Graph ----------")
+        mod = transform.AnnotateTarget(self.tidl_target)(mod)
+        #print(mod.astext(show_meta_data=False))
+        print("---------- Merge Compiler Regions ----------")
+        mod = transform.MergeCompilerRegions()(mod)
+        #print(mod.astext(show_meta_data=False))
+        print("---------- Partioned Graph ----------")
+        mod = transform.PartitionGraph()(mod)
+        #print(mod.astext(show_meta_data=False))
+        print("---------- Unpack composite ops in the graph ----------")
+        mod = UnpackComposites(mod, compiler=self.tidl_target)
+        #print(mod.astext(show_meta_data=False))
+        print('NUM TIDL SUBGRAPHS BEFORE PRUNING:', sum([1 for subgraph in mod.get_global_vars() if subgraph.name_hint.startswith(self.tidl_target)]))
+        print("---------- Prune Graph ----------")
+        mod = PruneSubgraphsWithMoreThanOneInput(mod, compiler=self.tidl_target)
+        #print(mod.astext(show_meta_data=False))
+        mod = PruneSubgraphs(mod, compiler=self.tidl_target, num_subgraphs_to_keep=self.num_tidl_subgraphs)
+        print('NUM TIDL SUBGRAPHS AFTER PRUNING:', sum([1 for subgraph in mod.get_global_vars() if subgraph.name_hint.startswith(self.tidl_target)]))
+        print(mod.astext(show_meta_data=False))

```

Comment:
Let's remove all print statements from this function

Reply:
Done